### DISCUSSION TITLE
BasicOTA with ESP32 and Arduino 2.0.0 rc3 = espota.py error

### DISCUSSION THREAD

**Discussion by tombo9999** [JAndrassy/ArduinoOTA#128] [Q&A]
I like BasicOTA because it's simple ;)
Tried on Windows 10 + Arduino IDE 1.8.19 and MacOS Monterey 12.1 + Arduino IDE 1.8.18 and it works without any problem.
Wanted to try Arduino 2.0.0 rc3 nightly build, can compile, but can't upload.
I'm just a hobbyist, not a programmer, surely I'm making some silly mistake.

Can someone give me a hint? Thanks in advance.

**_MacOS_**
Sketch uses 690349 bytes (52%) of program storage space. Maximum is 1310720 bytes.
Global variables use 41348 bytes (12%) of dynamic memory, leaving 286332 bytes for local variables. Maximum is 327680 bytes.

Compilation complete.
Usage: espota.py [options]

espota.py: error: option -p: invalid integer value: '{upload.port.properties.port}'
Upload error: Failed uploading: uploading error: exit status 2

**_Windows 10 (same error)_**
...
espota.exe: error: etc

**Comment by JAndrassy** [comment]
this is a different library than you use.

**Comment by tombo9999** [comment]
How to use your library? Beginner mistake?

Il giorno mer 5 gen 2022 alle ore 16:11 Juraj Andrássy <
***@***.***> ha scritto:

**Reply by JAndrassy** [reply]
don't use my library. the ArduinoOTA library bundled with the esp32 Arduino core is better for esp32. continue with it. but don't ask here about the problems with it

**Reply by tombo9999** [reply]
Sorry, understood only now.
The same filename made me confused :/
Sorry again
If you want you can delete this post.

Thanks

**Comment by JAndrassy** [comment]
I reproduced it and reported for IDE 2
https://github.com/arduino/arduino-ide/issues/740
